Makefile

```diff
@@ -81,8 +82,16 @@ $(SUBDIRS): force ## Execute default make target(make all) for the provided subd
 
 tests-privileged: ## Run Go tests including ones that require elevated privileges.
 	@$(ECHO_CHECK) running privileged tests...
-	PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) -p 1 $(TEST_LDFLAGS) \
-		$(TESTPKGS) $(GOTEST_BASE) $(GOTEST_COVER_OPTS) | $(GOTEST_FORMATTER)
+	## We split tests into two parts: one that can be run in parallel
+	## and tests that cannot be run in parallel with other packages
+	## Two drawbacks of this approach:
+	## - if first set of tests fails, second one is not run
```

Comment:
So coverage is still in the output of the job run via tparse at the end of "Runtime privileged tests" in CI: 
https://github.com/cilium/cilium/actions/runs/11213049806/job/31165676358
if both tests succeed, which is actually better than what we had before :sweat_smile: 

I decided not to do that (ipsec run in case of parallel tests failed), mostly because I would have to be very careful with handling error codes properly to ensure that the whole CI job fails if one of them fails.

Reply:
What do you think about the following? If we really want to execute it anyway despite the 1st set of parallel failing. Otherwise, if it is too unreadable/complex, I'm ok with sticking to the current.

```bash
tests-privileged: ## Run Go tests including ones that require elevated privileges.
	@$(ECHO_CHECK) running privileged tests...
	( PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) ${TESTPKGS} $(GOTEST_BASE) $(GOTEST_COVER_OPTS) ||  echo $$? > /tmp/errno1 )
	( PRIVILEGED_TESTS=true PATH=$(PATH):$(ROOT_DIR)/bpf $(GO_TEST) $(TEST_LDFLAGS) $(UNPARALLELTESTPKGS) $(GOTEST_BASE) -json -covermode=count -coverprofile=coverage2.out -p 1 --tags=unparallel || echo $$? > /tmp/errno2 )
	if [ -f /tmp/errno1 ]; then ERR=$$(cat /tmp/errno1) && rm -f /tmp/errno1 && exit $${ERR}; fi
	if [ -f /tmp/errno2 ]; then ERR=$$(cat /tmp/errno2) && rm -f /tmp/errno2 && exit $${ERR}; fi
	tail -n+2 coverage2.out >> coverage.out
	rm coverage2.out
	$(MAKE) generate-cov
```

Reply:
Resolving this comment: keeping this as potential follow-up and enhancement.